app/export.go

```diff
@@ -334,3 +340,73 @@ func (a *App) BuildPostReactions(postId string) (*[]ReactionImportData, *model.A
 	return &reactionsOfPost, nil
 
 }
+
+func (a *App) ExportCustomEmoji(writer io.Writer, file string) *model.AppError {
+	customEmojiList, err := a.GetEmojiList(0, 100, model.EMOJI_SORT_BY_NAME)
+	if err != nil {
+		return err
+	}
+
+	var dirName = "exported_emoji"
+
+	pathToDir := a.createDirForEmoji(file, dirName)
+
+	for _, emoji := range customEmojiList {
+		err := a.copyEmojiImages(emoji.Id, pathToDir)
+		if err != nil {
+			return model.NewAppError("BulkExport", "app.export.export_custom_emoji.copy_emoji_images.error", nil, "err="+err.Error(), http.StatusBadRequest)
+		}
+
+		emojiExportObject := &model.EmojiForExport{
+			Emoji: emoji,
+			Image: dirName + "/" + emoji.Id + "/image",
+		}
+		emojiImportObject := ImportCustomEmoji(emojiExportObject)
+
+		if err := a.ExportWriteLine(writer, emojiImportObject); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *App) createDirForEmoji(file string, dirName string) string {
+	pathToFile, _ := filepath.Abs(file)
+	pathSlice := strings.Split(pathToFile, "/")
+	if len(pathSlice) > 0 {
+		pathSlice = pathSlice[:len(pathSlice)-1]
+	}
+	pathToDir := strings.Join(pathSlice, "/") + "/" + dirName
+
+	if _, err := os.Stat(pathToDir); os.IsNotExist(err) {
+		os.Mkdir(pathToDir, os.ModePerm)
+	}
+	return pathToDir
+}
+
+func (a *App) copyEmojiImages(emojiId string, pathToDir string) error {
+	var err error
+	var emojiImagePath = "data/emoji/" + emojiId + "/image"
```

Comment:
Here we're assuming that the command `export bulk demo.json --all-teams` is being run from the root of `mattermost-server`. 

The problem I'm facing is, how do we get to the `data/emoji` folder if the export command is run from being inside any package of the repo. One solution would be to create a flag for the user to specify the relative or absolute path of the `data` folder.

Would like to know feedback on this.

Reply:
This is a tricky problem that we need a generic solution for across the whole Mattermost codebase, but which I believe we don't have currently. So I would suggest leaving it as-is and we can fix it here when we fix it everywhere else in the codebase to be working-directory independent.